### PR TITLE
Fix error caused by all_reduce call in domino

### LINKED ISSUE #6851: [QUESTIONS]：Some questions about running Domino
Hello, I have read the paper of Domino recently. And when I run the Domino according to this blog [https://github.com/microsoft/DeepSpeed/blob/master/blogs/deepspeed-domino/README.md](https://github.com/microsoft/DeepSpeed/blob/master/blogs/deepspeed-domino/README.md) and this one [https://github.com/microsoft/DeepSpeedExamples/blob/master/training/DeepSpeed-Domino/README.md](https://github.com/microsoft/DeepSpeedExamples/blob/master/training/DeepSpeed-Domino/README.md). I just run the shell [https://github.com/microsoft/DeepSpeedExamples/blob/master/training/DeepSpeed-Domino/pretrain_llama_7b.sh](https://github.com/microsoft/DeepSpeedExamples/blob/master/training/DeepSpeed-Domino/pretrain_llama_7b.sh). However, I met the following error:
Image
I think the cdb in deepspeed is from client such as Megatron? Therefore, I am very confused about the relationship between deepspeed and megatron. And if I want to run domino, should I use the [https://github.com/microsoft/DeepSpeedExamples/blob/master/training/DeepSpeed-Domino/](https://github.com/microsoft/DeepSpeedExamples/blob/master/training/DeepSpeed-Domino/) or Megatron-DeepSpeed. At the same time, according to the paper, domino is tensor-parallel only? I want to know whether domino supports zero3 or not since I found that the optimizer is from megatron not from deepspeed?

### PR DESCRIPTION
**Pull request by hwchen2017**
Fix #6851 
Initialize communication backend to fix error caused by all_reduce call in the Domino transformer layer. 
Verified correctness in local test.

### CODE REVIEW

File: deepspeed/runtime/domino/transformer.py [hunk 1]
```diff
@@ -249,6 +249,8 @@ def __init__(self,
                  output_bias=None):
         super(DominoTransformerLayer, self).__init__()
 
+        init_distributed()
```
Comment (GuanhuaWang) [inline deepspeed/runtime/domino/transformer.py:252]: maybe change this dist init() like this:

https://github.com/microsoft/DeepSpeed/blob/da771ed42e41a44d5047813ca4672f1cfe9d1731/deepspeed/runtime/zero/mics.py#L150-L152

**Review by GuanhuaWang — COMMENTED** [review]
thx @hwchen2017 please find comments

**Review by tjruwase — APPROVED** [review]

**Comment by hwchen2017** [comment]
> @hwchen2017 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
@microsoft-github-policy-service agree

**Comment by GuanhuaWang** [comment]
let's see if the OOM error still exists in our CI test `nv-torch-latest-v100`

**Pull request merged**